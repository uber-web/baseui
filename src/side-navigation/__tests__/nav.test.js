/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {mount} from 'enzyme';
import {Navigation, StyledNavItemContainer} from '../index.js';

const nav = [
  {
    title: 'Colors',
    subnav: [
      {
        title: 'Shades',
        itemId: '/',
        subnav: [
          {
            title: 'Light',
            itemId: '#level1.1.2.2',
          },
        ],
      },
    ],
  },
  {
    title: 'Sizing',
    itemId: '#level1.2',
  },
  {
    title: 'Typography',
    itemId: '#level1.3',
  },
];

describe('Side navigation', () => {
  it('renders expected number of nav items', () => {
    const wrapper = mount(<Navigation items={nav} />);
    expect(wrapper.find(StyledNavItemContainer)).toHaveLength(5);
  });
});
