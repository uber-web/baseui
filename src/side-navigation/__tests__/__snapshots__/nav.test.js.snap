// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side navigation renders titles correctly modified by mapItem 1`] = `
Array [
  <span>
    New 
    Colors
  </span>,
  <span>
    New 
    Shades
  </span>,
  <span>
    New 
    Light
  </span>,
  <span>
    New 
    Sizing
  </span>,
  <span>
    New 
    Typography
  </span>,
]
`;
